### PR TITLE
Search: fix widget setting not working for WordPress 5.8

### LINKED ISSUE #20055: Search: Widget settings are not accessible in upcoming widget editor
#### Steps to reproduce the issue

Start from a site using WP 5.8 Beta + Gutenberg, with a site using a Jetpack Search plan.

1. Go to Appearance > Customize > Widgets
2. Go to one of your widget areas.
3. Add a Jetpack Search widget. 
4. Try clicking the "Add filter" button.
5. Nothing happens.

Since the selectors have changed, the widget will need to be updated I believe. 

image

### PR DESCRIPTION
**Pull request by kangzj**
Fixes #20055

#### Changes proposed in this Pull Request:
* Changed the selector to determine whether the widget is Jetpack Search widget or not so that the widget setting works in WP 5.8.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Ensure widget setting still works under WP 5.7 (Jetpack Dev Env)
* Go to a WP 5.8 site with Jetpack Search subscription (I just upgraded my local WP using the beta tester plugin [here](https://wordpress.org/news/2021/06/wordpress-5-8-beta-1/))
* Open Customizer / Widget Editor
* Ensure you can add Jetpack Search widget and add/remove filters and successfully save settings.

Note: After testing this, I run `jetpack docker uninstall` and `jetpack docker install` to get the local env back - but all previous data would be lost, please do an export before upgrade to WP 5.8 if you'd need the local data.

### CODE REVIEW

**Comment by kangzj** [comment]
After fixing the issue, I performed a regression test on Search in WP 5.8. Everything works pretty smoothly. 

Only thing noticed was that the widget setting is not very stylish compared to in WP 5.7. But guess the widget will eventually be replaced by a Gutenberg search block, so seems to me not a thing to worry about at the moment.

File: projects/plugins/jetpack/modules/widgets/search/js/search-widget-admin.js [hunk 1]
```diff
@@ -300,10 +300,8 @@
 
 	// When widgets are updated, remove and re-add listeners
 	$( document ).on( 'widget-updated widget-added', function( e, widget ) {
-		widget = $( widget );
-
-		var id = widget.attr( 'id' ),
-			isJetpackSearch = ( id && ( -1 !== id.indexOf( 'jetpack-search-filters' ) ) );
+		var idBase = $( widget ).find('.id_base').val(),
```
Comment (bluefuton) [inline projects/plugins/jetpack/modules/widgets/search/js/search-widget-admin.js:303]: Does `.id_base` exist pre-WordPress 5.8 too?

Comment (kangzj) [reply projects/plugins/jetpack/modules/widgets/search/js/search-widget-admin.js:303]: Yes that's right. The field was added in version 2.x. Tested working in WP 5.7 as well. So it's a compatible way to use.

Comment (kangzj) [reply projects/plugins/jetpack/modules/widgets/search/js/search-widget-admin.js:303]: And WP 5.8 Widget Editor uses block but keeps the compatibility by adding these fields from the frontend
https://github.com/WordPress/gutenberg/blob/v9.8.3/packages/edit-widgets/src/blocks/legacy-widget/edit/dom-manager.js#L100-L131

**Review by jsnmoon — APPROVED** [review]
Excellent fix! Tests well in both 5.7 and 5.8-beta2.

**Comment by kangzj** [comment]
🚢 Shipped in r227487-wpcom

**Pull request merged**